### PR TITLE
Add HACKING.md

### PR DESCRIPTION
**Pull request by michalfabik**
A minimal version of build instructions. Nothing interesting here, it's just that every project uses a slightly different method and I keep forgetting and digging around looking for a HOWTO.

Signed-off-by: Michal Fabik <mfabik@redhat.com>

### CODE REVIEW

**Comment by mgrabovsky** [comment]
I'm not sure `HACKING.md` is the right name for this file. I'd search for this either in `BUILDING` or in the readme (which could use some modernisation anyway).

File: README [hunk 1]
```diff
@@ -26,3 +28,29 @@ its functionality is currently limited to Linux-based operating systems using
...
+
+## How to build Satyr from source
+
+### Install dependencies:
+
+    $ sudo dnf builddep --spec satyr.spec
+
+or
+
+    $ sudo ./autogen.sh sysdeps --install
```
Comment (mgrabovsky) [inline README:40]: The `sudo` here is considered harmful. `autogen.sh` invokes it when necessary.

**Comment by mgrabovsky** [comment]
I suppose you could also add a badge for the build action at the top, just to make it a little fancier, like this
```
![Build](https://github.com/abrt/satyr/workflows/Build/badge.svg)
```

**Comment by mgrabovsky** [comment]
One last thing: You need to add the `.md` extension to make sure the readme renders as Markdown.

**Comment by michalfabik** [comment]
> You need to add the .md extension to make sure the readme renders as Markdown  
  
That's precisely what I wanted to do first of all. But no, surely `README` on its own is a well-established convention and GitHub is smart enough to figure out Markdown anyway, better not tamper with it. Oh well.

**Review by mgrabovsky — APPROVED** [review]

**Pull request merged**